### PR TITLE
Upgrade pendulum to latest major version ~2.0

### LINKED ISSUE #9015: Update to Pendulum 2.1.0
**Description**

Currently, we use pendulum `1.4.4`, this is an old version, we should consider moving to `2.1.0`.

If we do this we should make it very public as it will break some jobs if users don't update names like `Interval` to `Duration`.

For pendulum changelog, [see here](https://pendulum.eustace.io/blog/category/releases/).

### PR DESCRIPTION
**Pull request by crhyatt**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]
I have no comments, but I would be happy if one more person would look at it.

File: airflow/models/dag.py [hunk 1]
```diff
@@ -412,15 +412,15 @@ def following_schedule(self, dttm):
...
+                following = dttm.in_timezone(self.timezone) + delta
             else:
                 # absolute (e.g. 3 AM)
                 naive = cron.get_next(datetime)
                 tz = pendulum.timezone(self.timezone.name)
                 following = timezone.make_aware(naive, tz)
             return timezone.convert_to_utc(following)
         elif self.normalized_schedule_interval is not None:
-            return dttm + self.normalized_schedule_interval
+            return timezone.convert_to_utc(dttm + self.normalized_schedule_interval)
```
Comment (ashb) [inline airflow/models/dag.py:423]: Why this change? This one feels like it shouldn't be needed.

**Review by potiuk — APPROVED** [review]

File: airflow/models/dag.py (hunk 1 above)
Comment (ashb) [reply airflow/models/dag.py:423]: (Like it's not a problem, just not sure why it's needed)

Comment (potiuk) [reply airflow/models/dag.py:423]: Comment: For me it was more of a consistency with the previous return. Agree it's not needed but you do not have to make the mental effort to check if the dttm is in the UTC already.

Comment (crhyatt) [reply airflow/models/dag.py:423]: Some of the test were failing because the function was not returning a tz aware datetime object.  Given the function is defined as returning UTC, I just forced the convention.
